### PR TITLE
Create simple security with user roles

### PR DESCRIPTION
**Pull request by Mattpeety**

### CODE REVIEW

File: src/main/java/pl/scoutCamp/api/controller/rest/LoginController.java [hunk 1]
```diff
@@ -8,24 +8,27 @@
 import org.springframework.security.core.Authentication;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-@RestController
-@AllArgsConstructor
-public class LoginController {
+//@RestController
```
Comment (Sokols) [inline src/main/java/pl/scoutCamp/api/controller/rest/LoginController.java:14]: Should this code be commented? If yes please add some TODO/FIXME and explain what should be done there.

File: src/main/java/pl/scoutCamp/infrastructure/security/SecurityConfiguration.java [hunk 2]
```diff
@@ -38,19 +39,21 @@ public AuthenticationManager authenticationManager(
...
-                        .anyRequest().authenticated());
+                        .requestMatchers("/login", "/error").permitAll()
+                        .requestMatchers("/regiment/regiments").hasAuthority("ADMIN")
+                        .anyRequest().authenticated())
+                .formLogin(AbstractAuthenticationFilterConfigurer::permitAll)
+                .logout(logout -> logout.logoutSuccessUrl("/login"));
         return http.build();
     }
 
+//public void configure(HttpSecurity http) throws Exception {
```
Comment (Sokols) [inline src/main/java/pl/scoutCamp/infrastructure/security/SecurityConfiguration.java:54]: same here

File: src/main/java/pl/scoutCamp/infrastructure/security/RoleEntity.java [hunk 3]
```diff
@@ -22,4 +25,7 @@ public class RoleEntity {
 
     @Column(name = "role")
```
Comment (Sokols) [inline src/main/java/pl/scoutCamp/infrastructure/security/RoleEntity.java:26]: I think "name" would be better here as "role" is the name for the whole object

File: src/test/java/pl/scoutCamp/infastructure/database/repository/jpa/TeamJpaRepositoryTest.java [hunk 4]
```diff
@@ -60,5 +60,56 @@ void thatTeamCanBeSavedCorrectly() {
         assertThat(availableTeams).hasSize(3);
     }
 
+<<<<<<< Updated upstream
```
Comment (Sokols) [inline src/test/java/pl/scoutCamp/infastructure/database/repository/jpa/TeamJpaRepositoryTest.java]: unresolved merge conflicts here I guess

File: src/main/java/pl/scoutCamp/infrastructure/security/RoleEntity.java [hunk 5]
```diff
@@ -6,13 +6,16 @@
...
+import java.util.Set;
 
 @Data
 @Builder
 @AllArgsConstructor
 @NoArgsConstructor
 @Entity
-@Table(name = "scoutcamp_role")
+@Table(name = "role")
 public class RoleEntity {
```
Comment (Sokols) [inline src/main/java/pl/scoutCamp/infrastructure/security/RoleEntity.java:19]: what about moving this class to "database/entity" folder to keep consistency?

**Comment by Mattpeety** [comment]
Changes will be added to the next commit.

**Pull request merged**